File: test_data/code/python/1002-find-common-characters.py
from collections import Counter
from typing import List

class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        result = []
        counter = Counter(words[0])
        
        for word in words[1:]:
            counter &= Counter(word)
            
        for char, count in counter.items():
            result.extend([char] * count)
            
        return result

File: test_data/code/java/1002-find-common-characters.java
import java.util.*;

class Solution {
    public List<String> commonChars(String[] words) {
        Map<Character, Integer> commonCount = new HashMap<>();
        for (String word : words) {
            Map<Character, Integer> count = new HashMap<>();
            for (char ch : word.toCharArray()) {
                count.put(ch, count.getOrDefault(ch, 0) + 1);
            }
            if (commonCount.isEmpty()) {
                commonCount = count;
            } else {
                for (Map.Entry<Character, Integer> entry : commonCount.entrySet()) {
                    commonCount.put(entry.getKey(), Math.min(commonCount.get(entry.getKey()), count.getOrDefault(entry.getKey(), 0)));
                }
            }
        }
        List<String> result = new ArrayList<>();
        for (Map.Entry<Character, Integer> entry : commonCount.entrySet()) {
            for (int i = 0; i < entry.getValue(); i++) {
                result.add(String.valueOf(entry.getKey()));
            }
        }
        return result;
    }
}